native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternResponseProcessor.java

```diff
@@ -0,0 +1,522 @@
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.flags.SymbolFlags;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.MapType;
+import io.ballerina.runtime.api.types.ObjectType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.ReferenceType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.utils.JsonUtils;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.utils.ValueUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import io.ballerina.stdlib.constraint.Constraints;
+import io.ballerina.stdlib.http.api.HttpUtil;
+import io.ballerina.stdlib.http.api.ValueCreatorUtils;
+import io.netty.handler.codec.http.HttpHeaders;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+import static io.ballerina.stdlib.http.api.HttpConstants.HTTP_HEADERS;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE_RESPONSE_BODY_FIELD;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE_RESPONSE_HEADERS_FIELD;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE_RESPONSE_MEDIA_TYPE_FIELD;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE_RESPONSE_STATUS_CODE_FIELD;
+import static io.ballerina.stdlib.http.api.HttpConstants.STATUS_CODE_RESPONSE_STATUS_FIELD;
+import static io.ballerina.stdlib.http.api.HttpErrorType.CLIENT_ERROR;
+import static io.ballerina.stdlib.http.api.HttpErrorType.INTERNAL_HEADER_BINDING_ERROR;
+import static io.ballerina.stdlib.http.api.HttpErrorType.PAYLOAD_BINDING_CLIENT_ERROR;
+
+public final class ExternResponseProcessor {
+
+    private static final String NO_HEADER_VALUE_ERROR_MSG = "no header value found for '%s'";
+    private static final String HEADER_BINDING_FAILED_ERROR_MSG = "header binding failed for parameter: '%s'";
+    private static final Map<String, String> STATUS_CODE_OBJS;
+
+    static {
+        Map<String, String> statusCodeObjectsMap = new HashMap<>();
+        statusCodeObjectsMap.put("100", "StatusContinue");
+        statusCodeObjectsMap.put("101", "StatusSwitchingProtocols");
```

Comment:
You can use following shorthand code:
```java
STATUS_CODE_OBJS = new HashMap<>() {{
  put("100", "StatusContinue");
  put("101", "StatusSwitchingProtocols");
}};
```

Reply:
Tried this:
```java
private static final Map<String, String> STATUS_CODE_OBJS = new HashMap<>() {{
   put("100", "StatusContinue");
   ...
}};    
```
SonarLint marks this as **noncompliant**:
- Only static class initializers should be used
-  Double Brace Initialization should not be used

Reply:
Ok. So, shall we try this then:
```java
   STATUS_CODE_OBJS = new HashMap<>();
   statusCodeObjectsMap.put("100", "StatusContinue");
   // other status codes
```

Reply:
I think, ur current implementation works fine.
+1